Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/ErrorHandler.java

```diff
@@ -114,27 +119,31 @@ public ErrorHandler(int errorHorizon, int forecastHorizon, int sequenceIndex, do
         int currentLength = (actualsFlattened == null) ? 0 : actualsFlattened.length;
         checkArgument(currentLength % inputLength == 0, "actuals array is incorrect");
         int forecastLength = (pastForecastsFlattened == null) ? 0 : pastForecastsFlattened.length;
-        checkArgument(forecastLength == currentLength * dimensions * 3 / inputLength, "misaligned forecasts");
+
         int arrayLength = max(forecastHorizon + errorHorizon, currentLength / inputLength);
         this.pastForecasts = new RangeVector[arrayLength];
         this.actuals = new float[arrayLength][inputLength];
 
         int length = forecastHorizon * inputLength;
+        checkArgument(forecastLength == currentLength * dimensions * 3 * forecastHorizon / inputLength,
```

Comment:
should this be forecastLength == currentLength * 3 * forecastHorizon / inputLength,

currentLength = (number of actual time steps stored) x inputLength and for each of the stored time steps we get a forecast whose length is forecastHorizon x inputLength (and then upper and lower for each, hence x 3)?

Reply:
added tests for input length = 2. It passed with current code. 

for each of the stored time steps we get a forecast whose length is forecastHorizon x inputLength x shingle size, right?

so forecastLength = number of actual time steps stored x forecastHorizon x inputLength x shingle size x 3
= currentLength x forecastHorizon x shingle size x 3
= currentLength x forecastHorizon x dimensions x 3 / inputLength

Reply:
Should we use following instead?

forecastLength = number of actual time steps stored x forecastHorizon x inputLength x 3
                          = currentLength x forecastHorizon x 3

Reply:
yes. Can we add the indentity in the comments (so that we do not have to think about it).

Reply:
See it. Thx.